projects/components/src/quick-search/quick-search.wo.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { BaseWidgetObject } from '../utils/test/widget-object/widget-object';
+import { DataUi } from './quick-search.dataui';
+
+const QUICK_SEARCH_CURRENT_RESULT = '.selected';
+
+/**
+ * Widget Object for Quick-Search
+ */
+export class QuickSearchWo<T> extends BaseWidgetObject<T> {
+    static tagName = 'vcd-quick-search';
+
+    /**
+     * Returns quick-search modal body
+     */
+    getModalBody = this.factory.dataUi(DataUi.modalBody);
+
+    /**
+     * Returns user search input
+     */
+    getInput = this.factory.dataUi(DataUi.searchInput);
+
+    /**
+     * Returns the titles of all sections appearing in the search results
+     */
+    getSearchResultSectionTitles = this.factory.dataUi(DataUi.searchResultSectionTitles);
+
+    /**
+     * Returns each search result item in each section
+     */
+    getSearchResultItems = this.factory.dataUi(DataUi.searchResultItems);
+
+    /**
+     * Returns the currently selected search result item
+     */
+    getSelectedSearchResultItem = this.factory.css(QUICK_SEARCH_CURRENT_RESULT);
+
+    /**
+     * Search results are loading if the spinner exists, that is, if it has length > 0.
+     * Search results are not loading if spinner has length 0.
+     */
+    getSpinner = this.factory.dataUi(DataUi.spinner);
+
+    /**
+     * Returns the element that shows "No results found" if there are no results
+     */
+    getNoResults = this.factory.dataUi(DataUi.noResults);
+
+    /**
+     * Accesses the page object behind the Quick Search modal
+     */
+    getModalBackdrop = this.factory.css('.modal-backdrop');
+
+    /**
+     * Returns the alerts to the user about their search input, if necessary
+     */
+    getSearchResultAlerts = this.factory.dataUi(DataUi.searchResultAlerts);
+
+    /**
+     * Returns the element (invisible to the user) that represents the top of the search results
```

Comment:
These elements are not invisible to the user. They display on the top and on the bottom of quick search.